pkg/controller/validation/account_validation_controller.go

```diff
@@ -0,0 +1,187 @@
+	awsv1alpha1 "github.com/openshift/aws-account-operator/pkg/apis/aws/v1alpha1"
+	"github.com/openshift/aws-account-operator/pkg/awsclient"
+	"github.com/openshift/aws-account-operator/pkg/controller/utils"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logf.Log.WithName("controller_account_validation")
+
+const (
+	controllerName = "accountvalidation"
+	moveWaitTime   = 5 * time.Minute
+)
+
+type ValidateAccount struct {
+	Client           client.Client
+	scheme           *runtime.Scheme
+	awsClientBuilder awsclient.IBuilder
+}
+
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	c, err := utils.NewControllerWithMaxReconciles(log, controllerName, mgr, r)
+	if err != nil {
+		return err
+	}
+
+	err = c.Watch(&source.Kind{Type: &awsv1alpha1.Account{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	reconciler := &ValidateAccount{
+		Client:           utils.NewClientWithMetricsOrDie(log, mgr, controllerName),
+		scheme:           mgr.GetScheme(),
+		awsClientBuilder: &awsclient.Builder{},
+	}
+
+	return utils.NewReconcilerWithMetrics(reconciler, controllerName)
+}
+
+// Retrieve all parents of the given awsId until the predicate returns true.
+func ParentsTillP(awsId string, client awsclient.Client, p func(s string) bool, parents *[]string) error {
+	listParentsInput := organizations.ListParentsInput{
+		ChildId: aws.String(awsId),
+	}
+	listParentsOutput, err := client.ListParents(&listParentsInput)
+	if err != nil {
+		return err
+	}
+	if len(listParentsOutput.Parents) == 0 {
+		log.Info("Exhausted search looking for root OU - root OU and account OU likely in separate subtrees.", "path", parents)
+		return nil
+	} else {
+		id := *listParentsOutput.Parents[0].Id
+		*parents = append(*parents, id)
+		if p(id) {
+			return nil
+		}
+		return ParentsTillP(id, client, p, parents)
+	}
+}
+
+// Verify if the account is already in the root OU
+// The predicate indicates if the parent considered the desired root was found.
+func IsAccountInRootOU(account awsv1alpha1.Account, client awsclient.Client, isRootOU func(s string) bool) bool {
+	if account.Spec.AwsAccountID == "" {
+		return false
+	}
+	parentList := []string{}
+	err := ParentsTillP(account.Spec.AwsAccountID, client, isRootOU, &parentList)
+	if err != nil {
+		return false
+	}
+	if len(parentList) == 1 {
+		return true
+	} else {
+		return false
+	}
+}
+
+func MoveAccount(account awsv1alpha1.Account, client awsclient.Client, targetOU string, dryRun bool) error {
```

Comment:
I guess the 'correct' way to configure that would be via an environment variable we can pass in - otherwise it will be backed into the binary forever, and once we want to change it, we have to trigger a complete deployment again - so I guess I would opt against compiling the feature flag in via a constant.

Not much of a 'flag' then IMO - after all - to reset it I now have to recompile the whole program.